### ISSUE TITLE
[Feature Request] Add environment variables in SBT to generated script

### ISSUE THREAD

**Issue by Stamper-repo** [xerial/sbt-pack#126]
Hi everyone!

Currently we are allowed to add JVM options but we cannot specify environment variables in SBT for the launch script. Would it be possible to add these?

Regards!

**Comment by xerial** [comment]
Interesting idea. I guess we need to add an option so that you can embed arbitrary commands (e.g., ``export SOME_ENV_VAR=xxxx`) before launching java command.

**Comment by Fathardie** [comment]
For the shell script this would not be necessary. If you call Java like this:

`ENV_VAR1=blabla ENV_VAR2="spaced blabla" java -cp ...`

The bat script is another situation due to Windows. That would be:

```
set ENV_VAR1=blabla
set ENV_VAR2="spaced blabla"
java -cp ...
```

The same single command environment variable syntax does not seem to be available on Windows:
https://superuser.com/questions/1049430/how-do-you-set-environment-variables-for-a-single-command-on-windows

But, if you prefer to keep it homogenuous, then `export ENV_VAR1=blabla` for linux works perfectly of course ^^

**Comment by xerial** [comment]
Good point. Then simply having packEnvVars := `Map("program name" -> Map("env var" -> "value"))` then we should expand them for shell script and windows bat file differently.

**Comment by Fathardie** [comment]
Yeah I think that would be great!
One more thing that would make it perfect:

Environment variables are called differently for Windows (%PATH%) and Linux ($PATH). Also you might want to set different environment variables for different OS's. Maybe it is handy to have 3 options:

```
packEnvVarsWindows := Map...
packEnvVarsLinux := Map...
packEnvVars := Map...
```
Where `packEnvVars` is commonly shared over both scripts?

**Comment by grogdotcom** [comment]
Any update on this?

**Comment by xerial** [comment]
Having a new setting like `packEnvVars := Map[String, Map[String, String]]` is ok for me. As I have other priorities, I'd like to have contributions in the form of PR.

**Comment by xerial** [comment]
packEnvVars is added, which will be available in sbt-pack 0.14
